.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUSTFLAGS: --deny warnings
+
+jobs:
+  backend:
+    name: 'Build & test backend'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: backend
+    steps:
+    - uses: actions/checkout@v2
+    - name: Prepare cache key
+      run: |
+        cp Cargo.lock cache-fingerprint
+        echo $RUSTFLAGS >> cache-fingerprint
+        echo $(rustc -V) >> cache-fingerprint
+    - name: Restore cargo cache
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cargo/registry
+          ~/.cargo/git
+          backend/target
+        key: ${{ runner.os }}-cargo-${{ hashFiles('backend/cache-fingerprint') }}
+        restore-keys: ${{ runner.os }}-cargo
+    - name: Build
+      run: cargo build --all-targets
+    - name: Test
+      run: cargo test
+    - name: Generate documenation
+      run: cargo doc --no-deps
+
+  frontend:
+    name: 'Build, lint & test frontend'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: frontend
+    steps:
+    - uses: actions/checkout@v2
+    - name: Restore NPM cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: ${{ runner.os }}-node-
+    - name: Installing dependencies (npm ci)
+      run: npm ci
+    - name: Build
+      run: npx webpack --mode=development --display-error-details
+    - name: Lint
+      run: npx eslint --max-warnings 0 src
```

Comment:
Should this use the `npm run` script? It doesn't have the specific flags you are using here, but is that even intentional?

Reply:
Yes it is. `--max-warning 0` is apparently the only way to forbid warnings on CI. But during development, you don't need the linter to fail for every warning. In particular, you might still be interested in the warning/error distinction of the linter.

Reply:
Don't you think this will lead to warnings being ignored?